internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go

```diff
@@ -51,9 +57,64 @@ func (m *DynamicScaleUpManager) Reconcile() []error {
 	glog.Infoln("running dynamic scale up reconcile event")
 	defer m.logReconcileEventEnd()
 
-	return nil
+	var errs []error
+	errs = m.reconcileClustersForRegions()
+
+	return errs
 }
 
 func (m *DynamicScaleUpManager) logReconcileEventEnd() {
 	glog.Infoln("dynamic scale up reconcile event finished")
 }
+
+// reconcileClustersForRegions creates an OSD cluster for each supported cloud provider and region where no cluster exists.
+func (m *DynamicScaleUpManager) reconcileClustersForRegions() []error {
+	var errs []error
+	glog.Infoln("reconcile cloud providers and regions")
+	var providers []string
+	var regions []string
+	status := api.StatusForValidCluster
+	//gather the supported providers and regions
+	providerList := m.ClusterProvidersConfig.ProvidersConfig.SupportedProviders
+	for _, v := range providerList {
+		providers = append(providers, v.Name)
+		for _, r := range v.Regions {
+			regions = append(regions, r.Name)
+		}
+	}
+
+	// get a list of clusters in Map group by their provider and region.
+	grpResult, err := m.ClusterService.ListGroupByProviderAndRegion(providers, regions, status)
+	if err != nil {
+		errs = append(errs, errors.Wrapf(err, "failed to find cluster with criteria"))
+		return errs
+	}
+
+	grpResultMap := make(map[string]*services.ResGroupCPRegion)
+	for _, v := range grpResult {
+		grpResultMap[v.Provider+"."+v.Region] = v
+	}
+
+	// create all the missing clusters in the supported provider and regions.
+	for _, p := range providerList {
+		for _, v := range p.Regions {
+			if _, exist := grpResultMap[p.Name+"."+v.Name]; !exist {
```

Comment:
the scale up detection logic will change

Reply:
This PR is about the refactor of moving the logic from one place to another.

Reply:
Thanks for clarifying.